src/main/java/com/exadel/sandbox/vacation/service/impl/VacationServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.exadel.sandbox.vacation.service.impl;
+
+import com.exadel.sandbox.exception.EntityNotFoundException;
+import com.exadel.sandbox.vacation.dto.VacationDto;
+import com.exadel.sandbox.vacation.entities.Vacation;
+import com.exadel.sandbox.vacation.repository.VacationRepository;
+import com.exadel.sandbox.vacation.service.VacationService;
+import lombok.RequiredArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class VacationServiceImpl implements VacationService {
+
+    private final VacationRepository vacationRepository;
+    private final ModelMapper mapper = new ModelMapper();
+
+    @Override
+    public List<VacationDto> getVacations() {
+        List<Vacation> vacations = vacationRepository.findAll();
+
+        return vacations.stream().map(vacation -> mapper.map(vacation, VacationDto.class)).collect(Collectors.toList());
+    }
+
+    @Override
+    public VacationDto getVacationById(Long id) {
+        Optional<Vacation> byId = vacationRepository.findById(id);
+
+        Vacation vacation = byId.orElseThrow(() -> new EntityNotFoundException("Vacation with id: " + id + " not found"));
+
+        return mapper.map(vacation, VacationDto.class);
+    }
+
+
+    @Override
+    public VacationDto create(VacationDto vacationDto) {
```

Comment:
about DTOs when we create something we send not all filed (ex: we do not send id, creationDate, modifiyngDate). 
For update we to should have another dto, because we can't update some fileds (ex: id, creationDate, modifiyngDate ).Let's create a structure like: someDTO, responseDTO (extends someDTO), createDTO (extends someDTO), updateDTO(extends someDTO).
It's okay if we get the same DTOs for example for creating and updating because maybe later new fields will be added and they will become different. Please change all places